### ISSUE TITLE
Question about editing the DB structure

### ISSUE THREAD

**Issue by darozak** [Darwin-River/Ex-Machinis#87]
Hi Carlos.  I'm trying to create a resource_effects entry that accepts a parameter off the stack for the affected drone.  This means that the resource_effects.drone field needs to accept a negative value per the [Design Manual](https://github.com/Darwin-River/Ex-Machinis/blob/develop/documents/mechanics.md#action-tables):

> Whenever a field in one of the static tables contains a value less than zero, it’s absolute value will be interpreted as a pointer (p) to the C++ variable variable array var(p), which contains values that were pulled from the stack and/or modified by the Game Engine.

However, resource_effects.drone is currently defined as an unsigned value in game-engine/sql/create_db.sql.  What is the process for changing this?  If I change UNSIGNED to SIGNED in create_db.sql, how do I propagate this into the active DB?  And is it possible to do so without loosing the dynamic user and spacecraft data that is already in the DB?

Basically, what is the best way to go about implementing changes to the DB structure?

**Comment by crodnun** [comment]
Hi Dave, 
What you are looking for is the mysql 'alter' command.
This command can be run at any DB and at  run time, just 'alters' properties of a given table.

Right now we have:

```
mysql> desc resource_effects;
+------------+---------------------+------+-----+---------+-------+
| Field      | Type                | Null | Key | Default | Extra |
+------------+---------------------+------+-----+---------+-------+
| id         | int(5) unsigned     | NO   | PRI | NULL    |       |
| drone      | int(5) unsigned     | YES  |     | NULL    |       |
| resource   | int(5) unsigned     | YES  |     | NULL    |       |
| protocol   | int(5) unsigned     | YES  | MUL | NULL    |       |
| event_type | int(2) unsigned     | YES  |     | NULL    |       |
| local      | tinyint(1) unsigned | NO   |     | 0       |       |
| installed  | tinyint(1) unsigned | NO   |     | 0       |       |
| locked     | tinyint(1) unsigned | NO   |     | 0       |       |
| deplete    | tinyint(1) unsigned | NO   |     | 0       |       |
| quantity   | int(5) unsigned     | YES  |     | NULL    |       |
| time       | int(5) unsigned     | YES  |     | NULL    |       |
+------------+---------------------+------+-----+---------+-------+
11 rows in set (0.00 sec)
```

So to alter any field you just need to run this command:

```
ALTER TABLE <table_name> MODIFY COLUMN <column_name> <datatype>;
``` 
Notice that <xxx> values above are just placeholders for the table, column and type you want to set.

<datatype> would be the new type, of course,

Hope it helps, anyways I can do the change for you if you want.

**Comment by darozak** [comment]
Thanks Carlos.  This helps. 

How do I ensure that any run-time changes I make to the DB are captured in the repository? Do I also need to manually update the change in create_db.sql and commit it to the repository?  

I'll try doing this myself.  I'd like to learn how this is done.

**Comment by crodnun** [comment]
Correct. It is what I normally do. Update tge sql file manually and push the changes.

**Comment by darozak** [comment]
Awesome! I'll give it a try.
